internal/kafka/internal/services/data_plane_cluster.go

```diff
@@ -243,23 +128,6 @@ func (d *dataPlaneClusterService) setClusterStatus(cluster *api.Cluster, status
 		metrics.UpdateClusterStatusSinceCreatedMetric(*cluster, api.ClusterReady)
 	}
 
-	if !remainingCapacity {
-		var desiredStatus api.ClusterStatus
-		if status.NodeInfo.Current >= d.getRestrictedCeiling(cluster, status) {
-			desiredStatus = api.ClusterFull
```

Comment:
The constants for `api.ClusterFull` and `api.ClusterComputeNodeScalingUp` are still defined in `pkg/api/cluster_types.go`.

Aside from that, if scaling mode is `auto` there is still logic that creates new OSD cluster nodes and deletes empty ones in `internal/kafka/internal/workers/clusters_mgr.go`.

Finally, the `HasAvailableCapacityInRegion` method returns true if autoscaling is enabled. There is also other logic that checks the scaling mode like in `reconcileClusterInstanceType`.

Is this intentional?

Reply:
The `api.ClusterFull` may still be used - i.e to mark the cluster as full once all the machine pools are full. 
The `api.ClusterComputeNodeScalingUp` one can be deleted as it won't be needed.

Reply:
As for the other logic, they are left over because there won't be much divergence but a some improvements on them to consider things like fleet isolation, not deleting a cluster but firing an alert etc